conda/activate.py

```diff
@@ -764,7 +764,7 @@ def native_path_to_unix(paths):  # pragma: unix no cover
     # cygpath.  When this happens, we have two different virtual POSIX machines, rooted at
     # different points in the Windows filesystem.  We do our path conversions with one and
     # expect the results to work with the other.  It does not.
-    from .common.path import which
+    from shutil import which
```

Comment:
Thinking about this a little more, we need to look into the differences between `distutils.spawn.find_exectuable` (which is what `conda.common.path.which` uses) and `shutil.which`:

https://github.com/python/cpython/blob/c1e71ce56fdb3eab62ad3190d09130f800e54610/Lib/shutil.py#L1452-L1530

vs

https://github.com/python/cpython/blob/d58ff6a616276c205e480225f588140f480271b7/Lib/distutils/spawn.py#L95-L129

In particular, `shutil.which` will:

- prepend `PATH` on Windows with the CWD
- will not only look for exe files but also the ones set in the `PATHEXT` env var, or when not set the list of `.COM;.EXE;.BAT;.CMD;.VBS;.JS;.WS;.MSC`
- runs access checks (if it exists and is executable) on the found path, which may have performance impacts on network drives

I just want to make sure we test these on Windows before committing to this change

Reply:
The Windows suite passes but we can run more things locally if needed. `find_executable` [also checks for existence](https://github.com/python/cpython/blob/d58ff6a616276c205e480225f588140f480271b7/Lib/distutils/spawn.py#L126), which is hopefully as slow on those cases as existence+access.

We _could_ add `os.curdir` to the end of PATH on purpose on Windows but I feel that would be more surprising than leaving it as is. I suspect there were strong reasons to add the default behaviour in `shutil.which`.

What kind of tests did you have in mind?

Reply:
Let's keep it the way you suggested, since I think `shutil` is a more complete approach than `distutils`' version, and it's also maintained.